crates/eframe/Cargo.toml

```diff
@@ -58,12 +58,15 @@ android-native-activity = ["egui-winit/android-native-activity"]
 ## If you plan on specifying your own fonts you may disable this feature.
 default_fonts = ["egui/default_fonts"]
 
+## Enable the directories dependency for a more precise file path for the persistence feature.
+## Not needed if you provide your own path, or if the simpler default path resolvers are good enough.
+directories = ["dep:directories"]
```

Comment:
Should this be a default feature to avoid breaking changes?

How often does the naive approach fail, and `directories` be required?

Reply:
The problem with making it default is that people using eframe without persistence would have the directories dependency even though it isn't being used for anything, unless they pass `--no-default-features`.

This naive function relies on default environment variables, so it will fail on custom environments that don't have those variables. I'm not sure how often that comes up for eframe users.

Another option would be to use the [home](https://crates.io/crates/home) crate (which eframe already depends on indirectly) to get the paths on Linux and macOS, and a modified version of [this code](https://github.com/rust-lang/cargo/blob/f2d1e37b56529f5200c3861be8b3d1c24f62c135/crates/home/src/windows.rs#L20-L42) to get the appdata path on Windows. I think that would completely reproduce what directories does, and then it could be removed or left as a dev dependency for the unit test.

Reply:
I made the change to just replicate what directories does entirely instead of using the naive approach. If it's too much unsafe/windows-specific code to be in eframe, we could use the etcetera crate instead, as suggested in the issue. Although it is a bit outdated with respect to the home crate at the moment.